src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp

```diff
@@ -1886,44 +1774,23 @@ nmethod* SharedRuntime::generate_native_wrapper(MacroAssembler* masm,
 
 #endif /* ASSERT */
 
-  // This may iterate in two different directions depending on the
-  // kind of native it is.  The reason is that for regular JNI natives
-  // the incoming and outgoing registers are offset upwards and for
-  // critical natives they are offset down.
+  // For JNI natives the incoming and outgoing registers are offset upwards.
   GrowableArray<int> arg_order(2 * total_in_args);
 
   VMRegPair tmp_vmreg;
   tmp_vmreg.set2(rbx->as_VMReg());
 
-  if (!is_critical_native) {
-    for (int i = total_in_args - 1, c_arg = total_c_args - 1; i >= 0; i--, c_arg--) {
-      arg_order.push(i);
-      arg_order.push(c_arg);
-    }
-  } else {
-    // Compute a valid move order, using tmp_vmreg to break any cycles
-    ComputeMoveOrder cmo(total_in_args, in_regs, total_c_args, out_regs, in_sig_bt, arg_order, tmp_vmreg);
```

Comment:
`ComputeMoveOrder` is still used somewhere, or?

Reply:
Yes, it's used in
cpu/x86/universalUpcallHandler_x86_64.cpp:  SharedRuntime::compute_move_order(in_sig_bt,

Reply:
FWIW, I have a change in panama-foreign repo that replaces that use with a custom class. Will remove ComputeMoveOrder there as well, and it should be completely gone after the next JEP integration, probably in 19.

Reply:
Ok, thanks Jorn.